opm/simulators/flow/EclWriter.hpp

```diff
@@ -375,7 +376,29 @@ class EclWriter : public EclGenericWriter<GetPropType<TypeTag, Properties::Grid>
         }
     }
 
-    void writeOutput(data::Solution&& localCellData, const SimulatorTimer& timer, bool isSubStep)
+    void writeReports(const SimulatorTimer& timer) {
+        const int reportStepNum = simulator_.episodeIndex() + 1;
+        auto rstep = timer.reportStepNum();
```

Comment:
I **think**, when written as this, we guarantee `rstep == reportStepNum`.  Might want to look into that at some point.

Reply:
Ah, always confused by the report steps.. Updated now, also fixed so that we do not get empty reports for wells that open later.

Reply:
>> I think, when written as this, we guarantee rstep == reportStepNum. Might want to look into that at some point.
>
>Ah, always confused by the report steps..

Everyone is.  This is a constant source of head scratching.  We desperately need better naming for this.

> Updated now, also fixed so that we do not get empty reports for wells that open later.

Much appreciated!